index.js

```diff
@@ -221,3 +221,84 @@ if (crypto) {
     activator
   );
 }
+
+var instrumentPromise = !!global.Promise;
+
+// Check that global Promise is native
+if (instrumentPromise) {
+  // shoult not use any methods that have already been wrapped
+  var promiseListener = process.addAsyncListener({
+    create: function create() {
```

Comment:
Are there promise implementations that are async? I can imagine any of the shims using process.nextTick to make their implementation "async" like the native one.